### PR TITLE
Pr 449 sagarrao

### PR DESCRIPTION
**Pull request by sfc-gh-japatel**
Open PR request: https://github.com/snowflakedb/snowflake-kafka-connector/pull/449

This is to push to release version 1.7.3

### CODE REVIEW

**Review by sfc-gh-tjones — APPROVED** [review]

File: src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnector.java [hunk 1]
```diff
@@ -59,6 +60,8 @@ public class SnowflakeSinkConnector extends SinkConnector {
   // Using setupComplete to synchronize
   private boolean setupComplete;
 
+  private static final long VALIDATION_NETWORK_TIMEOUT_MS = TimeUnit.SECONDS.toMillis(45);
```
Comment (sfc-gh-tzhang) [inline src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnector.java:63]: I thought you have added a few timeout values in another PR, what's the difference between this and the others?

File: src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnector.java [hunk 2]
```diff
@@ -231,7 +234,11 @@ public Config validate(Map<String, String> connectorConfigs) {
     SnowflakeConnectionService testConnection;
     try {
       testConnection =
-          SnowflakeConnectionServiceFactory.builder().setProperties(connectorConfigs).build();
+          SnowflakeConnectionServiceFactory.builder()
+              .setProperties(connectorConfigs)
+              .setNetworkTimeout(VALIDATION_NETWORK_TIMEOUT_MS)
```
Comment (sfc-gh-tzhang) [inline src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnector.java:239]: Should we added it to `connectorConfigs` or move this to the same place whether you added the other timeout constants? What's so special about this one?

File: src/test/java/com/snowflake/kafka/connector/internal/ConnectionServiceIT.java [hunk 3]
```diff
@@ -99,7 +99,7 @@ public void createConnectionService() {
         });
 
     SnowflakeURL url = TestUtils.getUrl();
-    Properties prop = InternalUtils.createProperties(TestUtils.getConf(), url.sslEnabled());
+    Properties prop = InternalUtils.createProperties(TestUtils.getConf(), url.sslEnabled(), 0);
```
Comment (sfc-gh-tzhang) [inline src/test/java/com/snowflake/kafka/connector/internal/ConnectionServiceIT.java:102]: could we add a test to verify non-zero timeout?

File: src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnector.java (hunk 1 above)
Comment (sfc-gh-japatel) [reply src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnector.java:63]: This timeout is for JDBC connection. https://docs.snowflake.com/en/user-guide/jdbc-parameters.html#networktimeout
Others were for httpclient which is used to call Snowpipe REST APIs.

File: src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnector.java (hunk 2 above)
Comment (sfc-gh-japatel) [reply src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnector.java:239]: Mentioned above. 
These timeouts is for checking if db exists, schema exists etc. (Connection through JDBC)

File: src/test/java/com/snowflake/kafka/connector/internal/ConnectionServiceIT.java (hunk 3 above)
Comment (sfc-gh-japatel) [reply src/test/java/com/snowflake/kafka/connector/internal/ConnectionServiceIT.java:102]: [sagarrao12](https://github.com/sagarrao12)
